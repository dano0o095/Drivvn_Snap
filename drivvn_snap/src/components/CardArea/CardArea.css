@keyframes animateUp {
  from {
    transform: translateY(100px);
  }
  to {
    transform: translateY(0px);
  }
}

#snap-text {
  animation: animateUp 0.5s linear 1;
}

