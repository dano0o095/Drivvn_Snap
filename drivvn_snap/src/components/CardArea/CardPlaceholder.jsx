import React from 'react'

const CardPlaceholder = () => {
  return (
    <svg data-testid="placeholder-svg" className="m-4 m-4" xmlns="http://www.w3.org/2000/svg" width="220" height="300" viewBox="0 0 220 300">
      <rect fill="#a8d4e0" width="220" height="300" />
    </svg>
  )
}

export default CardPlaceholder