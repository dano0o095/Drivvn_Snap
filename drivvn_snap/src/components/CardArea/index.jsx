import { connect } from 'react-redux'
import CardArea from './CardArea'
import { AddValueMatched, AddSuitMatched } from '../../store/cards/actions'

const mapStateToProps = state => ({
  newCard: state.cards.newCard,
  prevCard: state.cards.prevCard,
  totalValuesMatched: state.cards.matchedValues,
  totalSuitesMatched: state.cards.matchedSuits,
  remaningCardsCount: state.cards.remaining
})

const mapDispatchToProps = dispatch => ({
  addValueMatched: () => {
    dispatch(AddValueMatched())
  },
  AddSuitMatch: () => {
    dispatch(AddSuitMatched())
  },
})

export default connect(mapStateToProps, mapDispatchToProps)(CardArea)