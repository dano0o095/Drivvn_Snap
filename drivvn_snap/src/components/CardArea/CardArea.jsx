import React, { useEffect, useState } from 'react'
import propTypes from 'prop-types'

import api from '../../utils/api'
import CardPlaceholder from './CardPlaceholder'

const CardArea = props => {
  const { newCard, prevCard, addValueMatched, AddSuitMatch } = props

  const [matchType, setMatchType] = useState('')

  useEffect(() => {
    api.getNewDeck()
  }, [])

  useEffect(() => {
    if (newCard && Object.keys(newCard).length === 0 && newCard.constructor === Object) { // check if object exists
      setMatchType('')
    } else {
      if (newCard?.suit === prevCard?.suit) {
        setMatchType('SNAP SUIT!')
        AddSuitMatch()
      } else if (newCard?.value === prevCard?.value) {
        setMatchType('SNAP VALUE!')
        addValueMatched()
      } else {
        setMatchType('')
      }
    }
  }, [prevCard, newCard])

  return (
    <div className="container mb-5 mt-3">
      {matchType && <h4>{matchType}</h4>}
      {prevCard?.image && <img className="m-4 m-4" src={prevCard?.image} /> || <CardPlaceholder />}
      {newCard?.image && <img className="m-4 m-4" src={newCard?.image} /> || <CardPlaceholder />}
    </div>
  )
}

CardArea.propTypes = {

}

export default CardArea