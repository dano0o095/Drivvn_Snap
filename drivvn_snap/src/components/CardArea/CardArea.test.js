import React from 'react'
import * as reactRedux from 'react-redux'
import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import CardArea from './CardArea';

afterEach(cleanup)

it("renders without crashing", () => {
  const div = document.createElement('div')
  ReactDOM.render(<CardArea />, div)
})

it("matches snapshot of finished game", () => {
  React.useState = jest.fn()
    .mockReturnValueOnce(['', {}])
    .mockReturnValueOnce(['0 in 0', {}])
    .mockReturnValueOnce(['0 in 0', {}])

  const tree = renderer.create(
    <CardArea
      newCard={{ image: "https://deckofcardsapi.com/static/img/8C.png" }}
      prevCard={{ image: "https://deckofcardsapi.com/static/img/KH.png" }}
      remaningCardsCount={0}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})