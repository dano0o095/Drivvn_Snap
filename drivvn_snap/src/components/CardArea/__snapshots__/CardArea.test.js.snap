// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot of finished game 1`] = `
<div
  className="container mb-5 mt-3"
>
  
  <img
    className="m-4 m-4"
    data-testid="previous-card-image"
    src="https://deckofcardsapi.com/static/img/KH.png"
  />
  <img
    className="m-4 m-4"
    data-testid="new-card-image"
    src="https://deckofcardsapi.com/static/img/8C.png"
  />
  <h6
    data-testid="probability-suit"
  >
    0 in 0
     chance to match suit
  </h6>
  <h6
    data-testid="probability-value"
  >
    0 in 0
     chance to match value
  </h6>
</div>
`;
